pyro/infer/mcmc/mcmc.py

```diff
@@ -144,6 +144,11 @@ def _traces(self, *args, **kwargs):
         # restore original handler
         signal.signal(signal.SIGINT, sigint_handler)
         active_workers = self.num_chains
+        # To yield a deterministic ordering, we hold intermediate traces
+        # from each of the workers in its own queue in `results_buffer`
+        # and yield these in a round robin fashion.
+        cur_idx = 0
```

Comment:
nit: how about `current_chain` or `chain_idx`? I was a bit confused whether this indexed over workers or chains or time

Reply:
This is the round robin index, i.e. the index of the queue in the buffer we are currently blocked on and will be incremented with each `yield`. This might not be the same as the currently sampled chain, which is given by `chain_id`. Maybe we can call this `buffer_idx`?

Reply:
How are these numbers related: number of workers, number of chains, and number of buffers?

Reply:
They are the same - `chain_id` refers to the id of the chain/worker from where the last trace is sampled. `(chain_id, trace)` is sent as a tuple by each of the workers. As soon as this sample is received it is put into its own queue in the buffer. Then we keep yielding samples from the buffer (popping off the queue indexed by `buffer_idx`, which is incremented) until we are blocked, at which point we call `.get_nowait` on the shared queue again.